--- original/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
+++ patch/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
@@ -14,6 +14,7 @@
 import Nemo.DBus 2.0
 import com.jolla.lipstick 0.1
 import org.nemomobile.lipstick 0.1
+import org.nemomobile.configuration 1.0
 import "../backgrounds"
 import "../main"
 import "../statusarea"
@@ -125,9 +126,9 @@
         property int screenY: lockItem.contentTopMargin - clock.offset
                               + (clock.cannotCenter ? Theme.paddingLarge
                                                     : (lockItem.height - lockItem.contentTopMargin - height) / 2)
-        x: twoColumnMode ? screenY : (lockItem.width - width) / 2
-        y: twoColumnMode ? ((Screen.width - height) / 2)
-                         : screenY
+       x:(lockscreenUpcomingSettings.moveClock && !lockscreenUpcomingSettings.fixCalPos) ?(lockScreenPage.isPortrait?lockscreenUpcomingSettings.xPosClock:lockscreenUpcomingSettings.xPosLandscapeClock-clock.offset):(twoColumnMode ? screenY : (lockItem.width - width) / 2)
+        y:(lockscreenUpcomingSettings.moveClock && !lockscreenUpcomingSettings.fixCalPos) ?(lockScreenPage.isPortrait?lockscreenUpcomingSettings.yPosClock-clock.offset: lockscreenUpcomingSettings.yPosLandscapeClock)+lockItem.verticalOffset-lockItem.statusBarHeight  : (twoColumnMode ? ((Screen.width - height) / 2)
+                         : screenY)
         width: Math.max(clock.width, weatherIndicator.width)
         height: clock.height + weatherIndicator.height
 
@@ -230,6 +231,56 @@
             active: visible
         }
     }
+ConfigurationGroup {
+            id: lockscreenUpcomingSettings
+            path: "/desktop/lipstick-jolla-home/lockscreenUpcoming"
+            property bool fixCalPos: true
+            property int xPos: 0
+            property int yPos: 0
+            property int xPosLandscape: 0
+            property int yPosLandscape: 0
+            property bool moveClock: false
+
+            property int xPosClock: 0
+            property int yPosClock: 0
+            property int xPosLandscapeClock: 0
+            property int yPosLandscapeClock: 0
+        }
+        ShowCalEv
+        {
+           id: showcalevents
+           anchors 
+           {
+               top: lockscreenUpcomingSettings.fixCalPos ?(lockScreenPage.isPortrait?  centerBackground.bottom : centerBackground.top) : undefined
+               topMargin:lockscreenUpcomingSettings.fixCalPos ?  Theme.paddingMedium: undefined
+               right: lockscreenUpcomingSettings.fixCalPos ? parent.right: undefined
+               rightMargin: lockscreenUpcomingSettings.fixCalPos ?  Theme.paddingMedium+(lockScreenPage.isPortrait?0:clock.offset) : undefined
+            }
+           x: lockscreenUpcomingSettings.fixCalPos ? undefined: (lockScreenPage.isPortrait? lockscreenUpcomingSettings.xPos  :lockscreenUpcomingSettings.xPosLandscape -clock.offset) 
+          y : lockscreenUpcomingSettings.fixCalPos ? undefined: (lockScreenPage.isPortrait?lockscreenUpcomingSettings.yPos-clock.offset : lockscreenUpcomingSettings.yPosLandscape) +lockItem. verticalOffset-lockItem.statusBarHeight
+
+            opacity: Math.min(clock.transitionOpacity, clock.unlockOpacity)
+           state: "hidden"
+           states: [
+               State 
+               {
+                   name: "hinting"
+                   when: lockScreen.locked && !lockScreen.panning && lockItem.allowAnimations
+                   StateChangeScript
+                   {
+                       //name: "myScript"
+                       script:
+                       {
+                           showcalevents.runAnimations()
+                       }
+                   }                   
+               },
+               State
+               {
+                   name: "hidden"
+               }
+           ]
+        }
 
     ContrastBackground {
         id: mprisBackground
