--- original/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
+++ patch/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
@@ -15,6 +15,7 @@
 import com.jolla.lipstick 0.1
 import com.jolla.settings.system 1.0
 import org.nemomobile.lipstick 0.1
+import org.nemomobile.configuration 1.0
 import "../backgrounds"
 import "../main"
 import "../statusarea"
@@ -118,16 +119,57 @@
     ContrastBackground {
         id: centerBackground
 
-        x: (lockItem.width - width) / 2
+      //  x: (lockItem.width - width) / 2
         y: lockItem.contentTopMargin - clock.offset + (clock.cannotCenter
                 ? Theme.paddingLarge
                 : (lockItem.height - lockItem.contentTopMargin - height) / 2)
 
-        width: Math.max(clock.width, weatherIndicator.width)
-        height: clock.height + weatherIndicator.height
+        //width: Math.max(clock.width, weatherIndicator.width,showcalevents.width)
+      //  height: clock.height + weatherIndicator.height +showcalevents.height
 
+        width: lockItem.width
+        height:lockItem.height
         opacity: Math.min(clock.transitionOpacity, clock.unlockOpacity)
-
+ConfigurationGroup {
+            id: lockscreenUpcomingSettings
+            path: "/desktop/lipstick-jolla-home/lockscreenUpcoming"
+            property bool fixClock: true
+            property int xPos: 0
+            property int yPos: 0
+        }
+        ShowCalEv
+        {
+           id: showcalevents
+           anchors 
+           {
+               top: lockscreenUpcomingSettings.fixClock ?  weatherIndicator.bottom : undefined
+               topMargin:lockscreenUpcomingSettings.fixClock ?  Theme.paddingMedium: undefined
+               right: lockscreenUpcomingSettings.fixClock ? parent.right: undefined
+               rightMargin: lockscreenUpcomingSettings.fixClock ?  Theme.paddingMedium : undefined
+            }
+           x: lockscreenUpcomingSettings.fixClock ? undefined: lockscreenUpcomingSettings.xPos 
+          y : lockscreenUpcomingSettings.fixClock ? undefined: lockscreenUpcomingSettings.yPos         
+           state: "hidden"
+           states: [
+               State 
+               {
+                   name: "hinting"
+                   when: lockScreen.locked && !lockScreen.panning && lockItem.allowAnimations
+                   StateChangeScript
+                   {
+                       //name: "myScript"
+                       script:
+                       {
+                           showcalevents.runAnimations()
+                       }
+                   }                   
+               },
+               State
+               {
+                   name: "hidden"
+               }
+           ]
+        }
         Clock {
             id: clock
 
